app/components/post_list/new_messages_divider.js

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PropTypes} from 'react';
+import {StyleSheet, View} from 'react-native';
+
+import FormattedText from 'app/components/formatted_text';
+
+const style = StyleSheet.create({
+    container: {
+        alignItems: 'center',
+        flexDirection: 'row'
+    },
+    textContainer: {
+        marginHorizontal: 15
+    },
+    line: {
+        flex: 1,
+        height: 1,
+        backgroundColor: '#ffaf53',
+        opacity: 0.2
+    },
+    text: {
+        fontSize: 14,
+        fontWeight: '600',
+        color: '#ffaf53'
+    }
+});
+
+function NewMessagesDivider(props) {
+    return (
+        <View style={[style.container, props.style]}>
+            <View style={style.line}/>
+            <View style={style.textContainer}>
+                <FormattedText
+                    id='post_list.newMsg'
+                    defaultMessage='New Messages'
+                    style={style.text}
+                />
+            </View>
+            <View style={style.line}/>
+        </View>
+    );
+}
+
+NewMessagesDivider.propTypes = {
+    style: PropTypes.oneOfType([PropTypes.object, PropTypes.number])
```

Comment:
@enahum That's how a style(sheet) object is recognized in proptypes. Yeah, it's unintuitive. I think we should at some point extract it out to a separate file so there's a canonical declaration for it. Same with many of our types, especially for serialized records. We can discuss that later.

I used the `DateHeader` as a basis for this component. So that style proptype is already elsewhere in our codebase.